models/default_llm_backend.py

```
@@ -23,9 +22,31 @@
+    Google Cloud Vertex AI (via `VERTEX_AI_JSON`).  Automatically retries on
+    transient errors and compacts the conversation when the context window is
+    exceeded.
+
+    Attributes:
+        model (str): The litellm model string (e.g. `vertex_ai/claude-sonnet-4-6`).
+        max_tokens (int): Maximum tokens allowed in a single response.
+        api_key (str | None): API key for direct access; `None` when using Vertex AI.
+        vertex_credentials (str | None): JSON credentials for Vertex AI; `None` otherwise.
+    """
 
     def __init__(self, model: str, use_vertex_api: bool):
```

Ruff (0.15.21)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)